benches/benchmarks/toposort.rs

```diff
@@ -0,0 +1,73 @@
+#![allow(clippy::unit_arg)] // Required for black_box uses
+
+use criterion::{black_box, criterion_group, AxisScale, BenchmarkId, Criterion, PlotConfiguration};
+use portgraph::{
+    algorithms::{toposort, TopoSort},
+    Direction, NodeIndex, PortGraph,
+};
+
+use super::generators::*;
+
+fn run_toposort(graph: &PortGraph, roots: impl IntoIterator<Item = NodeIndex>) {
+    let topo: TopoSort = toposort(graph, roots, Direction::Outgoing);
+    for n in topo {
+        black_box(n);
+    }
+}
+
+#[cfg(feature = "petgraph")]
+fn run_petgraph_toposort(graph: &PortGraph) {
+    let topo: Vec<_> = petgraph::algo::toposort(&graph, None).unwrap();
+    for n in topo {
+        black_box(n);
+    }
+}
+
+fn bench_toposort(c: &mut Criterion) {
+    let mut g = c.benchmark_group("run toposort on a graph");
+    g.plot_config(PlotConfiguration::default().summary_scale(AxisScale::Logarithmic));
+
+    for size in [100, 1_000, 10_000] {
+        g.bench_with_input(BenchmarkId::new("toposort", size), &size, |b, size| {
+            let graph = make_two_track_dag(*size);
+            let roots = [0, 1].map(NodeIndex::new);
+            b.iter(|| black_box(run_toposort(&graph, roots)))
+        });
+    }
+    g.finish();
+}
+
+#[cfg(feature = "petgraph")]
+fn bench_petgraph_toposort(c: &mut Criterion) {
+    let mut g = c.benchmark_group("run petgraph's toposort on a graph");
+    g.plot_config(PlotConfiguration::default().summary_scale(AxisScale::Logarithmic));
+
+    for size in [100, 1_000, 10_000] {
+        g.bench_with_input(
+            BenchmarkId::new("toposort_petgraph", size),
+            &size,
+            |b, size| {
+                let graph = make_two_track_dag(*size);
+                b.iter(|| black_box(run_petgraph_toposort(&graph)))
+            },
+        );
+    }
+    g.finish();
+}
+
+#[cfg(feature = "petgraph")]
+criterion_group! {
+    name = benches;
+    config = Criterion::default();
+    targets =
+        bench_toposort,
+        bench_petgraph_toposort,
+}
+
+#[cfg(not(feature = "petgraph"))]
+criterion_group! {
+    name = benches;
+    config = Criterion::default();
```

Comment:
On my machine (which I assume to be average) for the larger cases I get the message:
```
Warning: Unable to complete 100 samples in 5.0s. You may wish to increase target time to 5.1s, enable flat sampling, or reduce sample count to 70.
```
so maybe the config could use a tweak?

Reply:
Ohh, that's annoying. In my laptop I get ~70ns per node for portgraph and ~55ns for petgraph, so the slowest test is `graph/toposort/10000` at 700.49 µs.

What timings are you getting? Is it from another bench? (You can run just these with `cargo bench topo --all-features`)

Reply:
Hmm, having run a few more times, my initial timeouts seem to be an aberration. Now I'm getting <1ms on graph/toposort/10000